internal/generationtest/generation_test.go

```diff
@@ -0,0 +1,76 @@
+package generationtest
+
+import (
+	"io/ioutil"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/bazelbuild/bazel-gazelle/testtools"
+	"github.com/bazelbuild/rules_go/go/tools/bazel"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// TestFullGeneration runs the gazelle binary on a few example
+// workspaces and confirms that the generated BUILD files match expectation.
+func TestFullGeneration(t *testing.T) {
+	tests := map[string][]bazel.RunfileEntry{}
+	runfiles, err := bazel.ListRunfiles()
+	if err != nil {
+		t.Fatalf("bazel.ListRunfiles() error: %v", err)
+	}
+	var manifest *manifestYAML
+	for _, f := range runfiles {
```

Comment:
Kind of silly to iterate through the runfiles twice. I think there's a better way.

Reply:
Eh, maybe not. Issue is we can't really pass the test anything because this version of go_test doesn't take env vars.